community/blog/2021-05 Enhanced request tracking in Arcus.WebApi v1.3/README.md

```diff
@@ -0,0 +1,135 @@
+        return Ok();
+    }
+}
+```
+
+When such a health endpoint is generated (like [Dapr](https://docs.dapr.io/developing-applications/building-blocks/observability/sidecar-health/)) and you have no control over where the this endpoint is hosted, you can't really add this attribute. For those scenarios we added an additional option to the configurable request tracking options:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+public class Startup
+{
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseRequestTracking(options => options.OmitRoutes.Add("/api/v1/health"));
+    }
+}
+```
+
+## Only track certain HTTP status codes
+
+As a continuation of the previous health endpoints example. It would be even better if we only could see health failures and disregard all success statuses. That would be both make the tracking more cleaner and help with defect detection when something's wrong.
+
+We created an `[RequestTracking]` attribute to pass along either a fixed HTTP status code or a range of HTTP status status codes.
+These status codes will act as an **allow list**. The request tracking will only happen when the HTTP response's status code is within the configured status code range.
+
+In the following example, the request tracking of the health endpoint will only happen when the HTTP response's status code is minimum 500 or maximum 599 (range is inconclusive).
+
+```csharp
+using Arcus.WebApi.Logging;
+using Microsoft.AspNetCore.Mvc;
+
+[ApiController]
+[Route("/api/v1/health")]
+public class HealthController : ControllerBase
+{
+    [HttpGet]
+    [RequestTracking(500, 599)]
+    public IActionResult Get()
+    {
+        // Determine health status...
+        return Ok();
+    }
+}
+```
+
+These HTTP status code ranges can also be specified in the global request tracking options. The request tracking will then only happen when **every** HTTP response's status code is within this specified range.
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+public class Startup
+{
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseRequestTracking(options => options.TrackedStatusCodeRanges.Add(new StatusCodeRange(500, 599));
+    }
+}
+```
+
+> Note that the `[RequestTracking]` attribute can be combined with the `TrackedStatusCodeRanges` option. The request tracking will then happen when the HTTP response's status is within both status code ranges.
+
+## Finer grained request/response body exclusion
+
+Tracking the HTTP request and/or response body can be quite expensive. Sometimes the request is worth tracking but the bodies are just to great to be included in the telemetry. We already provided in previous versions an request tracking option to opt-in for the inclusion of the HTTP request/response bodies, but this option acted on a global level. A finer control was required.
```

Comment:
What do you think?
```suggestion
Up until now, users had the choice of using request and/or response body tracking to gain more insights. However, this was only configurable for all operations which can become pretty expensive since that might be too much or does not give any added value.

Because of that, we are allowing you to choose where you want to track what!
```

Reply:
Yeah, is ok. Thanks!